executor/analyze.go

```diff
@@ -1440,19 +1507,29 @@ workLoop:
 					Count:     task.rootRowCollector.Base().Count - task.rootRowCollector.Base().NullCount[task.slicePos],
 					FMSketch:  task.rootRowCollector.Base().FMSketches[task.slicePos],
 					TotalSize: task.rootRowCollector.Base().TotalSizes[task.slicePos],
+					MemSize:   collectorMemSize,
 				}
 			}
 			if task.isColumn {
 				collectors[task.slicePos] = collector
 			}
+			releaseCollectorMemory := func() {
+				if !task.isColumn {
+					e.memTracker.Consume(-collector.MemSize)
+				}
+			}
 			hist, topn, err := statistics.BuildHistAndTopN(e.ctx, int(e.opts[ast.AnalyzeOptNumBuckets]), int(e.opts[ast.AnalyzeOptNumTopN]), task.id, collector, task.tp, task.isColumn)
 			if err != nil {
 				resultCh <- err
+				releaseCollectorMemory()
 				continue
 			}
+			finalMemSize := hist.MemoryUsage() + topn.MemoryUsage()
+			e.memTracker.Consume(finalMemSize)
```

Comment:
When do we release memory usage of hist and topn

Reply:
at the end of analyze, when detach, they are not big size.